components/org.wso2.carbon.identity.user.onboard.core.service/src/main/java/org/wso2/carbon/identity/user/onboard/core/service/password/ResetLinkGenerator.java

```diff
@@ -88,7 +97,32 @@ public String generateResetLink(Configuration configuration) throws IdentityReco
 
         String serverHost = ConfigurationFacade.getInstance().getAccountRecoveryEndpointPath();
 
-        return String.format("%s/confirmrecovery.do?confirmation=%s", serverHost, secretKey);
+        Boolean isDynamicAskPwdEnabled;
+        try {
+            isDynamicAskPwdEnabled = FlowMgtConfigUtils.getFlowConfig(Constants.FlowTypes.INVITED_USER_REGISTRATION.
+                            getType(), user.getTenantDomain()).getIsEnabled();
+        } catch (FlowMgtServerException e) {
+            throw new IdentityRecoveryServerException("Error while retrieving the flow configuration for " +
+                    "INVITED_USER_REGISTRATION flow.", e);
+        }
+
+        if (!isDynamicAskPwdEnabled) {
+            return String.format("%s/confirmrecovery.do?confirmation=%s", serverHost, secretKey);
+        }
+
+        String configuredPortalURL = null;
+        try {
+            BrandingPreferenceManager brandingPreferenceManager = new BrandingPreferenceManagerImpl();
+            configuredPortalURL = BrandingPreferenceMgtUtils.buildConfiguredPortalURL(null,
+                    configuration.getTenantDomain(), brandingPreferenceManager,
+                    Constants.FlowTypes.INVITED_USER_REGISTRATION.getType()
+            );
+        } catch (BrandingPreferenceMgtException | URLBuilderException e) {
+            LOG.error("Error retrieving configured portal URL for tenant: " + configuration.getTenantDomain(), e);
```

Comment:
If an error occurs, shouldn't we break the flow considering it can't be continued.